spec/derivative_services/external_metadata_derivative_service_spec.rb

```diff
@@ -57,6 +57,6 @@
     parent = query_service.find_by(id: parent_resource.id)
     expect(parent.title).to eq ["China census data by county, 2000-2010"]
     expect(parent.visibility).to eq ["open"]
-    expect(event_generator).to have_received(:record_updated).twice
+    expect(event_generator).to have_received(:record_updated).exactly(3).times
```

Comment:
Why did this change?

Reply:
I actually dug around this, and I have no idea. It seemed like it was firing everything at correct times, I'm not sure how it only fired twice before.

Reply:
I just dug through this again. There are 3 events:

1. The change_set is saved for the first time.
2. Saving the change_set for the first time fixes the visibility of the FileSet that's attached.
3. The metadata is extracted, firing the updated for the record again.